static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @typescript-eslint/no-misused-promises
+-- Fastdom measure and mutate signatures are Promise<void>
+-- Nested fastdom measure-mutate promises throw the error:
+-- "Promise returned in function argument where a void return was expected"
+*/
+import crossIcon from 'svgs/icon/cross.svg';
+import fastdom from '../../../../lib/fastdom-promise';
+
+const shouldRenderLabel = (adSlotNode: HTMLElement) =>
+	!(
+		adSlotNode.classList.contains('ad-slot--fluid') ||
+		adSlotNode.classList.contains('ad-slot--frame') ||
+		adSlotNode.classList.contains('ad-slot--gc') ||
+		adSlotNode.getAttribute('data-label') === 'false' ||
+		adSlotNode.getElementsByClassName('ad-slot__label').length
+	);
+
+const createAdCloseDiv = () => {
+	const closeDiv: HTMLElement = document.createElement('button');
+	closeDiv.className = 'ad-slot__close-button';
+	closeDiv.innerHTML = crossIcon.markup;
+	closeDiv.onclick = () => {
+		const container: HTMLElement | null = closeDiv.closest(
+			'.mobilesticky-container',
+		);
+		if (container) container.remove();
+	};
+	return closeDiv;
+};
+
+const createAdLabel = () => {
+	const adLabel = document.createElement('div');
+	adLabel.className = 'ad-slot__label';
+	adLabel.innerHTML = 'Advertisement';
+	adLabel.appendChild(createAdCloseDiv());
+	return adLabel;
+};
+
```

Comment:
Main changes are to the function `renderAdvertLabel` to either render the top banner as a toggled element or as previously - inserting dynamically.